### PR TITLE
fix: update to new font awesome css class

### LINKED ISSUE #2768: Usage of deprecated Font Awesome icon class
Looks like the `fa` icon class is [deprecated by Font Awesome v5.x](https://fontawesome.com/v5.15/how-to-use/on-the-web/referencing-icons/basic-use):

> The `fa` prefix has been deprecated in Version 5. The new default is the `fas` solid style and the `fab` style for brands.

For me it was hard to find the correct icon name I want to use in the [icons list](https://fontawesome.com/v5.15/icons?d=gallery&p=2&s=solid&m=free). In the end I used the v4.x [icon list](https://fontawesome.com/v4.7/icons/).

### PR DESCRIPTION
**Pull request by CFenner**
This updates the [deprecated](https://fontawesome.com/v5.15/how-to-use/on-the-web/referencing-icons/basic-use) `fa` css class to `fas` (solid type).

⚠️ This will probably break one or the other used custom icon which are not any more in [the *solid* icon set](https://fontawesome.com/v5.15/icons?d=gallery&p=2&s=solid&m=free). 

fixes #2768

### Visual Changes

| old | new |
| --- | --- |
| Bildschirmfoto 2022-01-05 um 13 37 09| Bildschirmfoto 2022-01-05 um 13 37 03 |
| Bildschirmfoto 2022-01-05 um 13 38 27 | Bildschirmfoto 2022-01-05 um 13 38 22 |
| Bildschirmfoto 2022-01-05 um 13 39 00 | Bildschirmfoto 2022-01-05 um 13 39 04 |
| Bildschirmfoto 2022-01-05 um 13 37 36 | Bildschirmfoto 2022-01-05 um 13 37 40|
| Bildschirmfoto 2022-01-05 um 13 41 12 | Bildschirmfoto 2022-01-05 um 13 41 04 |
|Bildschirmfoto 2022-01-06 um 22 12 00|Bildschirmfoto 2022-01-06 um 22 13 15|
|Bildschirmfoto 2022-01-06 um 22 37 03|Bildschirmfoto 2022-01-06 um 22 36 46|
|Bildschirmfoto 2022-01-06 um 22 38 36|Bildschirmfoto 2022-01-06 um 22 38 28|
|Bildschirmfoto 2022-01-06 um 22 39 33|Bildschirmfoto 2022-01-06 um 22 39 26|
|Bildschirmfoto 2022-01-06 um 22 39 40|Bildschirmfoto 2022-01-06 um 22 39 46|

TODO:
- [x] If it includes major visual changes please add screenshots.
- [x] Please run `npm run lint:prettier` before submitting so that style issues are fixed.
- [x] Don't forget to add an entry about your changes to the CHANGELOG.md file.
- [x] It is highly recommended that you update your branch of `develop` before creating a
pull request

### CODE REVIEW

File: modules/default/weather/current.njk [hunk 1]
```diff
@@ -15,7 +15,7 @@
                 {% if config.showWindDirection %}
                     <sup>
                         {% if config.showWindDirectionAsArrow %}
-                            <i class="fa fa-long-arrow-up" style="transform:rotate({{ current.windDirection }}deg);"></i>
+                            <i class="fas fa-long-arrow-up" style="transform:rotate({{ current.windDirection }}deg);"></i>
```
Comment (CFenner) [inline modules/default/weather/current.njk:18]:
```suggestion
                            <i class="fas fa-long-arrow-alt-up" style="transform:rotate({{ current.windDirection }}deg);"></i>
```

Bildschirmfoto 2022-01-05 um 13 37 36
Bildschirmfoto 2022-01-05 um 13 37 40

File: modules/default/currentweather/currentweather.js [hunk 2]
```diff
@@ -135,7 +135,7 @@ Module.register("currentweather", {
 			var windDirection = document.createElement("sup");
 			if (this.config.showWindDirectionAsArrow) {
 				if (this.windDeg !== null) {
-					windDirection.innerHTML = ' &nbsp;<i class="fa fa-long-arrow-down" style="transform:rotate(' + this.windDeg + 'deg);"></i>&nbsp;';
+					windDirection.innerHTML = ' &nbsp;<i class="fas fa-long-arrow-down" style="transform:rotate(' + this.windDeg + 'deg);"></i>&nbsp;';
```
Comment (CFenner) [inline modules/default/currentweather/currentweather.js:138]:
```suggestion
					windDirection.innerHTML = ' &nbsp;<i class="fas fa-long-arrow-alt-down" style="transform:rotate(' + this.windDeg + 'deg);"></i>&nbsp;';
```
Bildschirmfoto 2022-01-05 um 13 41 12
Bildschirmfoto 2022-01-05 um 13 41 04

File: modules/default/clock/clock.js [hunk 3]
```diff
@@ -230,13 +230,13 @@ Module.register("clock", {
 			moonWrapper.innerHTML =
 				'<span class="' +
 				(isVisible ? "bright" : "") +
-				'"><i class="fa fa-moon-o" aria-hidden="true"></i> ' +
+				'"><i class="fas fa-moon-o" aria-hidden="true"></i> ' +
```
Comment (CFenner) [inline modules/default/clock/clock.js:233]:
```suggestion
				'"><i class="fas fa-moon" aria-hidden="true"></i> ' +
```
Bildschirmfoto 2022-01-05 um 13 38 27
Bildschirmfoto 2022-01-05 um 13 38 22

File: modules/default/clock/clock.js [hunk 4]
```diff
@@ -199,13 +199,13 @@ Module.register("clock", {
 			sunWrapper.innerHTML =
 				'<span class="' +
 				(isVisible ? "bright" : "") +
-				'"><i class="fa fa-sun-o" aria-hidden="true"></i> ' +
+				'"><i class="fas fa-sun-o" aria-hidden="true"></i> ' +
```
Comment (CFenner) [inline modules/default/clock/clock.js:202]:
```suggestion
				'"><i class="fas fa-sun" aria-hidden="true"></i> ' +
```

Bildschirmfoto 2022-01-05 um 13 37 09
Bildschirmfoto 2022-01-05 um 13 37 03

**Comment by MichMich** [comment]
Thanks! Seems good to me. Could you add an updated changelog?

**Comment by CFenner** [comment]
Just noticed that the new `calendar` icon is also different. We could change to `calender-alt` that looks a bit as the one before.

Bildschirmfoto 2022-01-06 um 22 13 15

**Comment by khassel** [comment]
test fails seems to be unrelated to this PR, tried a fix for this in https://github.com/MichMich/MagicMirror/pull/2778

so can be merged from my side

**Comment by MichMich** [comment]
I merged @khassel's PR, and now this one has some conflicts. Sorry. Is it possible for you to fix this?

**Comment by khassel** [comment]
probably the CHANGELOG as usual ... 😜

**Comment by MichMich** [comment]
I'm currently using the GitHub app. Unfortunately that doesn't show the conflicting files. I'll check it tomorrow on my desktop.

**Comment by CFenner** [comment]
@MichMich ready to merge 💫

**Comment by MichMich** [comment]
🎉

**Pull request merged**